Zend/zend_alloc.c

```diff
@@ -51,6 +51,7 @@
  * with more specialized routines when the requested size is known.
  */
 
+#include "php.h"
```

Comment:
This whole area is not my domain of expertise, but from a quick glance I really don't understand why you added a ``#include "php.h"`` here? Especially as that one should *not* be used anywhere within ``Zend/`` as it is for external extension use only AFAIK.

Reply:
This was added because I guard the `zend_mm_observer_register()` function to only be called during PHP module startup.
```c
ZEND_ASSERT(php_during_module_startup());
```
Maybe there is another way to do this

Reply:
I think in general the PHP code just... assumes you don't do the wrong thing. I may have to agree with George here: php.h should not really be included in zend_alloc.c.

Reply:
I removed the `php.h` and the corresponding check.